explainer.md

```diff
@@ -53,7 +51,13 @@ See individual sections for more details on these characteristics.
 
 ## Type definitions
 
-The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of a value.
+The central part of the Typed Objects specification are *type definition objects*, generally called *type definitions* for short. Type definitions describe the fixed structure of a value, and are used to specify a struct type's fields' types.
+
+A type definition has an internal method `[[PerformTypeCheck]]`, which, given a value `V`, returns a new value `vResult` which is of the right type, or throws.
```

Comment:
Could there be two internal methods:
* `[[Write]] : jsval &rarr; whatever`
* `[[Read]] : whatever &rarr; jsval`
?  Otherwise, as is, the `jsval` becomes the canonical representation of the typed object field and the wasm JS API spec will have an awkward time working with this directly, particularly for all the integral types.

Reply:
Alternatively, it might be useful to give Typed Objects a link back to their Typed Definition Object (for an immutable `.constructor` so instanceof works) and then `[[FieldTypes]] could be found via `o.[[TypeDef]].[[FieldTypes]]`.